include/linux/page_compat.h

```diff
@@ -75,4 +77,18 @@
 
 #endif /* ZFS_GLOBAL_ZONE_PAGE_STATE */
 
+/* Linux 5.2 made wait_on_page_writeback() GPL only */
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 2, 0)
```

Comment:
We're going to need to add an configure check for this, see the `ZFS_AC_KERNEL_BIO_SET_DEV_GPL_ONLY` check in `config/kernel-bio_set_dev.m4` for an example.

Although, what might be simpler in this particular case is to update `zfs_putpage` to directly call `wait_on_page_bit()` since it's the only caller.  Then we don't need any compatibility code.

Reply:
> We're going to need to add an configure check for this, see the `ZFS_AC_KERNEL_BIO_SET_DEV_GPL_ONLY` check in `config/kernel-bio_set_dev.m4` for an example.

Added.

> 
> Although, what might be simpler in this particular case is to update `zfs_putpage` to directly call `wait_on_page_bit()` since it's the only caller. Then we don't need any compatibility code.

It now only has `wait_on_page_writeback()` in `module/zfs/zfs_vnops.c` for GPL export case.

Reply:
Actually, that's not quite what I was suggesting, but after re-reading my comment I see why it was confusing.  We could absolutely go with your updated patch.  However, since there's only this one caller I was thinking we could do away with the configure check entirely and do something much simpler, like this:

```diff
@@ -4526,8 +4526,10 @@ zfs_putpage(struct inode *ip, struct page *pp, struct wri
                unlock_page(pp);
                rangelock_exit(lr);
 
-               if (wbc->sync_mode != WB_SYNC_NONE)
-                       wait_on_page_writeback(pp);
+               if (wbc->sync_mode != WB_SYNC_NONE) {
+                       if (PageWriteback(pp))
+                               wait_on_page_bit(pp, PG_writeback);
+               }
 
                ZFS_EXIT(zfsvfs);
                return (0);
```

Reply:
Oh.
I could change to do like that, but it would start to be more complex if there are >1 consumers of it.

Reply:
This code actually hasn't changed in quite a while, and offhand I can't think of any pending work where we'd need to add another caller.  So why don't we keep it simple for now and avoid the wrapper entirely.  As you said, it's only two lines after all.

Reply:
Updated.